2023/incoming/issue-388.pod

```diff
@@ -0,0 +1,313 @@
+Author: oodler@cpan.org
+Title: St. Nick's Reindeers Need More H2O!
+Topic: Util::H2O::More 
+
+=encoding utf8
+
+Last year in 2022, Santa Clause discovered a very useful Perl module called
+L<Util::H2O> that makes it very seamless to add I<object oriented programming>
+conveniences to Perl programs; it is particularly useful when it comes to
+adding Perl OOP to existing or legacy code bases.
+
+Over the last year, he's done quite a bit of digging into a related module
+named, L<Util::H2O::More>. The following is a summary of a few neat things
+available in this module, most of which is built upon L<Util::H2O>'s C<h2o>
+method which gives acessors to to C<HASH> references - or the I<inverse>,
+that converts an I<blessed> reference with accessors into a pure C<HASH>
+reference, C<o2h>.
+
+=head2 Santa's Got Options ... for Options
+
+There are lots of modules on CPAN that present L<Getopt::Long> in different
+ways, L<Util::H2O::More> makes its entry into the mix with two different
+methods; one that builds upon the other.
+
+One is a nifty method called C<opt2h2o>. This method makes it more
+convenient to define the fields of a reference destined to be objectified
+by C<h2o> by using the parameter array description format used by
+C<Getopt::Long::GetOptionsFromArray>.
+
+C<opt2h2o> is used to provide a list of default accessors to C<h2o> to an given
+C<HASH> reference. This reference is then provided to C<GetOptionsFromArray>,
+as shown in the example below that can explain better than mere words:
+
+=begin perl
+  use strict;
+  use warnings;
+  use Util::H2O::More qw/h2o opt2h2o/;
+  use Getopt::Long qw//;
+  
+  my @opts = (qw/name=s country=s meaning=s/);
+  my $o = h2o { meaning => q{???} }, opt2h2o(@opts);     # example showing how to set default
+  Getopt::Long::GetOptionsFromArray( \@ARGV, $o, @opts );
+  
+  if ($o->country) {
+    printf qq{Did you know, Santa is known as '%s' in the country/region of %s? It means, "%s"!\n}, $o->name, $o->country, $o->meaning;
+  }
+=end perl
+
+Which results in,
+
+=begin text
+  prompt> perl Santa-facts.pl --name Jultomten --meaning "Christmas Brownie" --country Sweden
+  Did you know, Santa is known as 'Jultomten' in the country/region of Sweden? It means, "Christmas Brownie"!
+=end text
+
+But as famous I<elvish> pitchman, Willy Nays says, I<but wait, there's
+more ...>
+
+L<Util::H2O::More> has a method that combines everything, including the need
+to C<require Getopt::Long;>, called C<Getopt2h2o>; it can dramatically shorten
+the amount of typing needed to process options I<and> provide accessors to
+the options:
+
+=begin perl
+  use strict;
+  use warnings;
+  use Util::H2O::More qw/Getopt2h2o/;
+
+  my $o = Getopt2h2o \@ARGV, { meaning => q{???} }, qw/name=s country=s meaning=s/;
+
+  if ($o->country) {
+    printf qq{Did you know, Santa is known as '%s' in the country/region of %s? It means, "%s"!\n}, $o->name, $o->country, $o->meaning;
```

Comment:
Do you break up the line because it's too long for the HTML styling?

Reply:
updated, lmk

Reply:
> Do you break up the line because it's too long for the HTML styling?

We've had some issues with longer lines on smaller screens, so I think it's probably safer to have less width where it's not required.